crates/turbopack-ecmascript/src/references/constant_value.rs

```diff
@@ -35,17 +35,20 @@ impl CodeGenerateable for ConstantValue {
         _context: Vc<Box<dyn EcmascriptChunkingContext>>,
     ) -> Result<Vc<CodeGeneration>> {
         let value = self.value.clone();
-        let visitors = [
-            create_visitor!(exact &self.path.await?, visit_mut_expr(expr: &mut Expr) {
-                *expr = match value {
-                    CompileTimeDefineValue::Bool(true) => quote!("(\"TURBOPACK compile-time value\", true)" as Expr),
-                    CompileTimeDefineValue::Bool(false) => quote!("(\"TURBOPACK compile-time value\", false)" as Expr),
-                    CompileTimeDefineValue::String(ref s) => quote!("(\"TURBOPACK compile-time value\", $e)" as Expr, e: Expr = s.to_string().into()),
-                };
-            }),
-        ]
-        .into();
+        let path = &self.path.await?;
 
-        Ok(CodeGeneration { visitors }.cell())
+        let visitor = create_visitor!(path, visit_mut_expr(expr: &mut Expr) {
+            *expr = match value {
+                CompileTimeDefineValue::Bool(true) => quote!("(\"TURBOPACK compile-time value\", true)" as Expr),
+                CompileTimeDefineValue::Bool(false) => quote!("(\"TURBOPACK compile-time value\", false)" as Expr),
+                CompileTimeDefineValue::String(ref s) => quote!("(\"TURBOPACK compile-time value\", $e)" as Expr, e: Expr = s.to_string().into()),
+                CompileTimeDefineValue::JSON(ref s) => quote!("(\"TURBOPACK compile-time value\", JSON.parse($e))" as Expr, e: Expr = s.to_string().into()),
```

Comment:
Is this used somewhere? Can we hoist this value into a module-level constant and refer to it?

Reply:
what do you mean?

Reply:
Currently, we have something like:

```js
function foo() {
  return JSON.parse('{"foo": true').foo
}
```

It'd be better if we hoisted that JSON constant:
```js
const __FOO__ = JSON.parse('{"foo": true');
function foo() {
  return __FOO__.foo
}
```

Reply:
I guess it would be nice, but also kinda annoying to do with how our codegen stuff currently works

webpack also injects everything directly

Reply:
the module level constant would still be mutable on the inside, which could break things as well 🤔